merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/GetSimulationResultsAction.java

```diff
@@ -162,14 +162,14 @@ public Map<String, List<Violation>> getViolations(final PlanId planId)
         .orElseGet(Collections::emptyMap);
     final var discreteProfiles = new HashMap<String, DiscreteProfile>(_discreteProfiles.size());
     for (final var entry : _discreteProfiles.entrySet()) {
-      discreteProfiles.put(entry.getKey(), DiscreteProfile.fromSimulatedProfile(entry.getValue().getRight()));
+      discreteProfiles.put(entry.getKey(), DiscreteProfile.fromSimulatedProfile(entry.getValue().getRight().stream().map($ -> Pair.of($.extent(), $.dynamics())).toList()));
```

Comment:
Better to update `fromSimulatedProfile` and `fromExternalProfile` instead of converting to the old format, because with this change all use sites will be using the new format.

Reply:
The issue I ran into there is that `fromSimulatedProfile` is defined in the `constraints` subproject, which doesn't depend on the `merlin-driver` subproject (which is where I put the `ProfileSegment` record).

Options are:
- Convert between the two in `merlin-server`, which depends on both subprojects
- Make one depend on the other
- Put `ProfileSegment` in a different subproject that both `constraints` and `merlin-driver` depend on.

There is one subproject that fits the bill for the last point - `merlin-sdk`(i.e. `merlin.protocol`) - though its intended purpose is to provide an interface between the mission model and the merlin-driver - not between the merlin-driver and downstream processes... But I could convince myself that resource profiles are a central enough concept to warrant membership in the `merlin.protocol`.

Reply:
Since `merlin.protocol` is provided by the runtime environment and not bundled with the model, I could see an argument for putting it there. But I don't really think it's something a model ever needs, so I'm not really a fan of that argument.

On the other hand, what if we just merged `constraints` into `merlin-driver`? Are there any clients of one that aren't already clients of the other? And if not, do we expect it to be a burden?

Reply:
I like putting it in merlin-sdk. I think putting it next to the types that it's made of (`Duration`, `RealDynamics`, `SerializedValue`) makes a lot of sense.

Reply:
I'm piecing together that `ProfileSegment` needs to depend on `merlin-sdk` (as you pointed out, Joel, it is composed of pieces from there), but that it's a downstream detail that should not be made available to mission models (as you pointed out, Jonathan).

Merging `constraints` into `merlin-driver` is a bit bigger of a change than I was hoping to make in this PR, so I'll take the middle ground of making `constraints` depend on `merlin-driver`. We've had to duplicate everything to do with `SimulationResults` between the two, and this discussion is highlighting that anew. I'd be interested in discussing further where our "lines in the sand" are best drawn beyond the context of this PR - perhaps a Github discussion (or a thread in public slack 🥺) is warranted.

I'll report back once I've made the change

Reply:
For ease of review, I've made the changes in a separate commit: https://github.com/NASA-AMMOS/aerie/commit/5c990bf9b7bbbaed550b6cb9aa290eafe4763adb